### PR TITLE
remove module version number from `<scoped-name>`

### LINKED ISSUE #18: remove module version number from `<scoped-name>`
We include the module version number in message types, property names and capability names, e.g. `core1.set` rather than `core.set`. I've been thinking, and I don't see any actual reason for that. What it might do, however, is make backwards-compatible implementations harder: When a client says `(want core1 core2)`, it always has to keep track of which version was agreed to by the server, even if it only uses features which remain identical across major versions, so that it can form the correct message types. Also, it may mean that a backwards-compatible client needs more roundtrips than necessary to negotiate capabilities, e.g.

```
(want core1 core2) ---> wait for reply -+-> (want core1.some-capability) ---> wait for reply
                                        |
                                        `-> (want core2.some-capability) ---> wait for reply
```

instead of:

```
(want core1 core2 core.some-capability) ---> wait for reply
```

I suggest that we remove the module version number from `<scoped-name>`, e.g. `core1.set` becomes `core.set`. I'm putting this as a question issue rather than a PR since this leads to changes in every module, and I want to make sure we agree on this before I actually invest the time to make the edits.

### PR DESCRIPTION
**Pull request by majewsky**
The reason for this is mostly YAGNI. I don't see any disadvantages to this change, and two main advantages:

1. Backwards-compatible implementations become easier. For example, once there are three different major versions of vt6/core, some clients might only want to use those behaviors that stayed the same across all those major versions. Those applications can just `(want core1 core2 core3)` and check that the server agrees to *any* version of core. Then they can forget about the negotiation entirely, because the message types and other scoped names are the same regardless of which major version of the module is in use.

2. It makes talking about versions easier. Before this change, we used at least the following identifiers for vt6/core1.0:

       vt6/core1.0    in the spec title
       vt6/core1      in the backwards-compatible parts of the spec
       vt6/core       to refer to the module regardless of major version
       core1.0        in the spec text
       core1          in scoped names and in the spec text
       core           in the spec text

   This change eliminates all usages of "core1" and "vt6/core1" except for the want message (where it is actually required), so the only remaining identifiers are:

       vt6/core1.0    to refer to a spec
       vt6/core       to refer to a module
       core           in scoped names and in the spec text

   The only exception is in `have` and `want` messages, where "core1" and "core1.0" are entirely appropriate.

Closes #18.

### CODE REVIEW

File: spec/core/1.0.md [hunk 1]
```diff
@@ -259,16 +257,23 @@ For each object (message type, property or capability) in a module, that module
...
+major-version = version-number
+minor-version = version-number
+full-version  = major-version "." minor-version
+```
+
+Each specification of a module has a **major version number** and a **minor version number**, as accepted by the `<major-version>` and `<minor-version>` grammar elements defined above.
+When both version numbers are shown together, they are formatted as implied by the definition of the `<full-version>` grammar element defined above.
 
-When a new module specification is created, its major and minor version number MUST be set to 1 and 0, respectively.
+When the first specification of a module is created, its major and minor version number MUST be set to 1 and 0, respectively.
```
Comment (laerling) [inline spec/core/1.0.md:271]: "When the specification of a new module is [first] created" would be a more elegant wording. Just my two cents.
(The "first" being optional, there)

File: spec/core/1.0.md [hunk 2]
```diff
@@ -353,17 +358,21 @@ Quoted strings are not allowed in this message.
...
+module-name-with-major-version = module-name major-version
+```
+
 The `want` message informs the server of modules and capabilities that the client wants to use.
-Each argument is a module name as accepted by `<module-name>`, or a capability name as accepted by `<message-type>`, except for `want` and `have`.
-For each argument that is a capability name, the server MUST already have agreed to using the module that defines this capability.
-Alternatively, one of the arguments that comes before this argument MUST be that module's name.
+Each argument is a pair of module name and major version number as accepted by `<module-name-with-major-version>`, or a capability name as accepted by `<scoped-name>`.
+For each argument that is a capability name, the server MUST already have agreed to using a module that defines this capability.
+Alternatively, one of the arguments that comes before this argument MUST be a pair of that module's name and a major version such that there exists a specification for that module and major version defining this capability.
```
Comment (laerling) [inline spec/core/1.0.md]: Add "as accepted by `<module-name-with-major-version>`" between "version" and "such", as in line 366.

File: spec/core/1.0.md [hunk 3]
```diff
@@ -394,12 +403,12 @@ From this and the definition of the `want` message also follows that the use of
 For `want` arguments that are capability names, the corresponding `have` argument is the same string, the capability name.
 A server SHALL NOT agree to using a capability if it has not previously agreed to using the module defining said capability, unless it agrees to using the module in the same `have` message.
 
-For `want` arguments that are module names, the corresponding `have` argument is the module name, with a dot and a positive integer appended to it.
+For `want` arguments that are pairs of module names and major version, the corresponding `have` argument is the corresponding `want` argument, with a dot and a positive integer appended to it.
```
Comment (laerling) [inline spec/core/1.0.md]: "Corresponding" used two times.
Use e. g. "For each `want` argument that is a pair of module names and major version, the corresponding `have` argument is that same `want` argument".

**Review by laerling — APPROVED** [review]
Little wording mishaps. Please consider changes according to my notes.

**Comment by majewsky** [comment]
I've addressed two of three comments. The remaining comment addresses a bigger issue, which I've filed as #24 since I want to move this change forward.

**Pull request merged**